features/step_definitions/machine.rb

```diff
@@ -5,6 +5,11 @@
     logger.warn "Not an IPI deployment, there are no machines"
     logger.warn "We will skip this scenario"
     skip_this_scenario
+    else if (machines[3] == nil)
```

Comment:
@jhou1 , with `machine.length` it takes the number of master machines into account as well . When masters are increased beyond 3 then only failure will occur will be simpler to debug if at all that happens .
I wanted to have a simple condition to avoid runnning cases if no workers are present , due to profile or any scenario etc.

Reply:
@jhou1 , do you think below will be good 

`  if !(machines.to_s.include?'worker')
       logger.warn "there are no worker machines,skipping the scenarios to test on workers"
       skip_this_scenario
  end
`
instead of the above conditions ?

Reply:
@miyadav I'm confused, can you please clarify IPI without workers? is it missing the worker machineset or a worker machineset is present but is scaled to 0? Your code needs to represent that accordingly.